src/go/k8s/cmd/configurator/main.go

```diff
@@ -109,32 +129,46 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
-	hostIndex, err := hostIndex(c.hostName)
+	podOrdinal, err := hostIndex(c.hostName)
 	if err != nil {
 		log.Fatalf("%s", fmt.Errorf("unable to extract host index: %w", err))
 	}
 
-	log.Printf("Host index calculated %d", hostIndex)
+	log.Printf("Host index calculated %d", podOrdinal)
 
-	err = registerAdvertisedKafkaAPI(&c, cfg, hostIndex, kafkaAPIPort)
+	err = registerAdvertisedKafkaAPI(&c, cfg, podOrdinal, kafkaAPIPort)
 	if err != nil {
 		log.Fatalf("%s", fmt.Errorf("unable to register advertised Kafka API: %w", err))
 	}
 
 	if cfg.Pandaproxy != nil && len(cfg.Pandaproxy.PandaproxyAPI) > 0 {
 		proxyAPIPort := getInternalProxyAPIPort(cfg)
-		err = registerAdvertisedPandaproxyAPI(&c, cfg, hostIndex, proxyAPIPort)
+		err = registerAdvertisedPandaproxyAPI(&c, cfg, podOrdinal, proxyAPIPort)
 		if err != nil {
 			log.Fatalf("%s", fmt.Errorf("unable to register advertised Pandaproxy API: %w", err))
 		}
 	}
 
-	cfg.Redpanda.ID = int(hostIndex)
+	restCfg, err := k8sConfig.GetConfig()
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to create kubernetes rest config: %w", err))
+	}
 
-	// First Redpanda node need to have cleared seed servers in order
-	// to form raft group 0
-	if hostIndex == 0 {
-		cfg.Redpanda.SeedServers = []config.SeedServer{}
+	k8sClient, err := client.New(restCfg, client.Options{})
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to create kubernetes client: %w", err))
+	}
+
+	err = calculateRedpandaID(func() int {
+		return int(rand.NewSource(time.Now().Unix()).Int63())
```

Comment:
so it's random and not incrementing as we discussed somewhere on slack? I would strongly preferred to do that :)